### ISSUE TITLE
lowercase of response header names

### ISSUE THREAD

**Issue by zakshevskii** [n-fuse/pajax#1]
I suggest to add lowercasing of header names. I discovered that firefox returns me 'Content-Type' header, but in Chrome it's 'content-type'.

**Comment by Dani723** [comment]
http headers are case-insensitive
http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

**Comment by OrKoN** [comment]
I think that they are case case-insensitive matters for the http protocol but pajax module should handle them better. As far as I understand, the issue is that response headers exposed by the pajax module are not consistent. For example, one cannot check response headers with a simple check:

```
 if (response['Content-Type'] == 'my-type') {

}
```
Because it can be response['content-type']. Is it correct? @zakshevskii 

I think pajax is intended to hide such differences. E.g by normalizing header names or by exposing getters to read headers. Does it make sense?

**Comment by Dani723** [comment]
I've looked into jquerys ajax function. The returned response headers are also case sensitive.
e.g. in Firefox it is "Content-Type" vs "content-type" in chrome.

But I absolutely see your point.
How about just lowercasing all header field names when using the "responseHeaders" option?

so it is always `req.responseHeaders['content-type']`.

**Comment by OrKoN** [comment]
Pajax should be better than jQuery.ajax :-) So I think always lower-casing is fine.

**Comment by OrKoN** [comment]
I think Jquery exposes jqXhr object which provides methods to get response headers https://api.jquery.com/jQuery.ajax/#jqXHR

**Comment by OrKoN** [comment]
https://github.com/jquery/jquery/blob/9d1b989f20b550af3590691723b0620f6914626e/src/ajax.js#L424

**Comment by Dani723** [comment]
Ah, ok. I've only tested the getAllResponseHeaders() function.
But I don't like to overload pajax with functions and helpers. I would simply modify the parsing, so there are only lowercased header fields.

**Comment by vanthome** [comment]
+1
